packages/launch/src/focused-launch/summary/index.tsx

```diff
@@ -289,25 +289,22 @@ const PlanStep: React.FunctionComponent< PlanStepProps > = ( {
 
 	const { defaultPaidPlan, defaultFreePlan, planPrices } = usePlans();
 
-	// persist selected plan if it's paid in order to keep displaying it in the
-	const onceSelectedPaidPlan = React.useRef( selectedPaidPlan );
+	// persist non-default selected paid plan if it's paid in order to keep displaying it in the plan picker
+	const [ nonDefaultPaidPlan, setNonDefaultPaidPlan ] = React.useState< Plan | undefined >();
+
+	React.useEffect( () => {
+		if ( selectedPaidPlan && selectedPaidPlan?.storeSlug !== defaultPaidPlan?.storeSlug ) {
+			setNonDefaultPaidPlan( selectedPaidPlan );
+		}
+	}, [ selectedPaidPlan, defaultPaidPlan, nonDefaultPaidPlan ] );
 
 	const isPlanSelected = ( plan: Plan ) => plan && plan.storeSlug === selectedPlan?.storeSlug;
 
 	const { sitePlan } = useSite();
 
-	const nonDefaultPaidPlan =
-		onceSelectedPaidPlan?.current &&
-		defaultPaidPlan &&
-		onceSelectedPaidPlan?.current?.storeSlug !== defaultPaidPlan?.storeSlug
-			? onceSelectedPaidPlan?.current
-			: undefined;
-
-	const allAvailablePlans: ( Plan | undefined )[] = [
-		defaultPaidPlan,
-		nonDefaultPaidPlan && onceSelectedPaidPlan.current,
-		defaultFreePlan,
-	];
+	const allAvailablePlans = [ defaultPaidPlan, nonDefaultPaidPlan, defaultFreePlan ].filter(
+		Boolean
+	) as Plan[];
```

Comment:
If all plans are undefined at this point, the `allAvailablePlans` array won't have any element and with the current approach, we're using this Array to show correct number of placeholders.
Issue with screen recordings: https://github.com/Automattic/wp-calypso/issues/48621
PR with a proposed fix: https://github.com/Automattic/wp-calypso/pull/48620